pkg/cfn/manager/tasks.go

```diff
@@ -1,158 +1,215 @@
 package manager
 
 import (
+	"fmt"
+	"strings"
 	"sync"
 
-	"k8s.io/apimachinery/pkg/util/sets"
-
 	"github.com/kris-nova/logger"
 
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha4"
 )
 
-type taskFunc func(chan error, interface{}) error
+// Task is a common interface for the stack manager tasks
+type Task interface {
+	Do(chan error) error
+	Describe() string
+}
 
-// Task has a function with an opaque payload
-type Task struct {
-	Call taskFunc
-	Data interface{}
+// TaskTree wraps a set of tasks
+type TaskTree struct {
+	tasks    []Task
+	Parallel bool
+	DryRun   bool
+	Sub      bool
```

Comment:
What is Sub?

Reply:
A sub-task, i.e. child not a parent. It's just so we can print user-friendly messages. Would `Child` be better? Besides, please do review the message formatting, it should evident from the tests and the readme changes.

Reply:
maybe something like `isChild`?

Reply:
or isSubtask?

Reply:
Let's go with `isSubtask`, cause that doesn't imply changing words that are used in output.